gtsfm/averaging/translation/averaging_creta.py

```diff
@@ -0,0 +1,119 @@
+            absolute_pose_priors: Priors on the camera poses (not delayed).
+            i2Ti1_priors: Priors on the pose between camera pairs (not delayed) as (i1, i2): i2Ti1.
+            scale_factor: Non-negative global scaling factor.
+            gt_wTi_list: List of ground truth poses (wTi) for computing metrics.
+
+        Returns:
+            Global camera poses wTi. The number of entries in the list is `num_images`. The list
+                may contain `None` where the global translations could not be computed (either underconstrained system
+                or ill-constrained system).
+            A GtsfmMetricsGroup with translation averaging metrics.
+            Indices of inlier measurements (list of camera pair indices).
+        """
+        # Generate CReTA inputs.
+        """[From CReTA_demo.m]
+        The data should contain the following variables:
+        Graph has N nodes and M edges.
+        RT: 3XM matrix of relative translation directions (Tij=Rj*(Ti-Tj)).  [i2Ui1 in our notation]
+        edges: Mx2 matrix of camera pairs.RT
+        matches: Mx1 cell each containing Kx4 matrix.
+          Each matrix contains the point correspondences for each edge.
+          Each row in the matrix contains (xi,yi,xj,yj) where xi,yi and xj,yj
+          are coordinates for images i,j respectively afer correcting for camera intrinsics.
+        R_avg: 3X3XN matrix of absolute rotations.  [iRw in our notation]
+        maxImages: Maimum No. of images in the dataset (used for indexing)
+        NOTE: Ensure that the graph is connected and has the maximal parallel
+        rigid component.
+        """
+        w_i2Ui1_dict, valid_cameras = get_valid_measurements_in_world_frame(i2Ui1_dict, wRi_list)
+        RT = np.array([i2Ui1.point3().flatten() for i2Ui1 in i2Ui1_dict.values()]).T
+        R_avg = np.array([wRi_list[ii].matrix() for ii in valid_cameras]).T
+
+        edges = np.array([(i1, i2) for i1, i2 in i2Ui1_dict.keys()])
+        matches = []
+        for (i1, i2), i2Ui1 in i2Ui1_dict.items():
+            if i2Ui1 is None:
+                continue
+            # Calibrate keypoints.
+            uv_norm_i1 = feature_utils.normalize_coordinates(keypoints_list[i1].coordinates, intrinsics[i1])
+            uv_norm_i2 = feature_utils.normalize_coordinates(keypoints_list[i2].coordinates, intrinsics[i2])
+
+            corr = corr_idxs_dict[(i1, i2)]
+            matches_i12i = np.hstack((uv_norm_i1[corr[:, 0]], uv_norm_i2[corr[:, 1]]))
+            matches.append(matches_i12i)
+        matches_hash = cache_utils.generate_hash_for_numpy_array(np.concatenate(matches))
+
+        # Write inputs for MATLAB script.
+        savemat(f"ta_input_{matches_hash}.mat", {"RT": RT, "edges": edges, "R_avg": R_avg, "matches": matches})
+        if not os.path.isfile(f"ta_output_{matches_hash}.mat"):
+            raise FileExistsError("Could not find CReTA translation averaging solution.")
+
```

Comment:
could we add a comment for L98 about what needs to be run in between L97 and L100?